packages/ui/src/context/translation-context.tsx

```diff
@@ -0,0 +1,32 @@
+import { createContext, useContext } from 'react'
+
+import { i18n, TFunction } from 'i18next'
+
+export interface TranslationStore {
+  t: TFunction
+  i18n: i18n
+  changeLanguage: (lng: string) => void
+}
+
+export const TranslationContext = createContext<TranslationStore | null>(null)
+
+export const TranslationProvider = ({
+  children,
+  useTranslationStore
+}: {
+  children: React.ReactNode
+  useTranslationStore: () => TranslationStore
+}) => {
+  const store = useTranslationStore()
+  return <TranslationContext.Provider value={store}>{children}</TranslationContext.Provider>
+}
+
+export const useTranslation = () => {
+  const context = useContext(TranslationContext)
+
+  if (!context) {
+    throw new Error('useTranslation must be used within TranslationProvider')
```

Comment:
If there is no context, we need to be fault-tolerant and instead return a function that just returns the fallback value for the string. See example in the commebnt above.